scalanet/test/src/io/iohk/scalanet/peergroup/kademlia/XorSpec.scala

```diff
@@ -0,0 +1,47 @@
+package io.iohk.scalanet.peergroup.kademlia
+
+import io.iohk.scalanet.peergroup.kademlia.Generators._
+import io.iohk.scalanet.peergroup.kademlia.Xor._
+import org.scalacheck.Gen.posNum
+import org.scalatest.FlatSpec
+import org.scalatest.Matchers._
+import org.scalatest.prop.GeneratorDrivenPropertyChecks._
+import scodec.bits.BitVector
+
+class XorSpec extends FlatSpec {
```

Comment:
I would like to ask to add the unidirectionality property (page 4, last paragraph of section 2.1):
- For any key x and distance D > 0, there is exactly one key y such that distance(x,y) = D.

The test could be implemented as follows: forall x, y and z with y != z then d(x, y) != d(x, z) (which fails with the current metric).

Reply:
Added. Thanks.